frontend/model/contracts/group.js

```diff
@@ -203,65 +307,64 @@ DefineContract({
           })
         })
       }),
-      process ({ data, meta }, { state }) {
+      process ({ data, meta }, { state, getters }) {
         // TODO: checkpointing: https://github.com/okTurtles/group-income-simple/issues/354
         const initialState = merge({
           payments: {},
+          paymentsByMonth: {},
           invites: {},
           proposals: {}, // hashes => {} TODO: this, see related TODOs in GroupProposal
           settings: {
             groupCreator: meta.username
           },
           profiles: {
             [meta.username]: initGroupProfile(meta.identityContractID, meta.createdDate)
-          },
-          userPaymentsByMonth: {
-            [currentMonthTimestamp()]: initMonthlyPayments()
           }
         }, data)
         for (const key in initialState) {
           Vue.set(state, key, initialState[key])
         }
+        initFetchMonthlyPayments({ meta, state, getters })
       }
     },
     'gi.contracts/group/payment': {
-      validate: objectOf({
+      validate: objectMaybeOf({
         toUser: string,
         amount: number,
-        currency: string,
+        currency: string, // must be one of the keys in currencies.js (e.g. USD, EUR, etc.)
```

Comment:
Can't we have a checker like `oneOf(['USD', 'EUR'])` or `oneOf(Object.keys(currencies))`?

Reply:
In the next update I plan to change this to:

```js
currencyFromTo: tupleOf(string, string), // must be one of the keys in currencies.js (e.g. USD, EUR, etc.)
```

The reason why I'm keeping it as `string, string` is because `currencies.js` will be constantly updated. Doing `oneOf(Object.keys(currencies))` would break clients that are out of date when one of the clients is updated with new currencies.